server/modules/analysis/service/_rChain/_files/system/rFileReadData.js

```diff
@@ -22,28 +26,67 @@ export default class RFileReadData extends RFileSystem {
     await super.init()
 
     const survey = this.rChain.survey
+    const surveyId = Survey.getId(survey)
     const cycle = this.rChain.cycle
     const steps = ProcessingChain.getProcessingSteps(this.rChain.chain)
 
-    const schema = SchemaRdb.getName(Survey.getId(survey))
+    const schema = SchemaRdb.getName(surveyId)
 
     for (const step of steps) {
       if (ProcessingStep.hasEntity(step)) {
         const entityDef = Survey.getNodeDefByUuid(ProcessingStep.getEntityUuid(step))(survey)
         const entityDefParent = Survey.getNodeDefParent(entityDef)(survey)
         const viewName = NodeDefTable.getViewName(entityDef, entityDefParent)
-        const fields = ProcessingStep.getCalculations(step).reduce((fieldsAcc, calculation) => {
-          const nodeDefUuid = ProcessingStepCalculation.getNodeDefUuid(calculation)
-          const nodeDef = Survey.getNodeDefByUuid(nodeDefUuid)(survey)
+        const calculationAttrDefs = R.pipe(
+          ProcessingStep.getCalculations,
+          R.map(
+            R.pipe(ProcessingStepCalculation.getNodeDefUuid, nodeDefUuid =>
+              Survey.getNodeDefByUuid(nodeDefUuid)(survey),
+            ),
+          ),
+        )(step)
+
+        const fields = ['*']
+        const fromTables = [viewName]
+        for (const nodeDef of calculationAttrDefs) {
           const nodeDefName = NodeDef.getName(nodeDef)
-          return `${fieldsAcc}, uuid_generate_v4() as ${nodeDefName}_uuid`
-        }, '*')
-        const selectData = dbGetQuery(schema, viewName, fields, [`${DataTable.colNameRecordCycle} = '${cycle}'`])
+          // Add nodeDefName_uuid field
+          fields.push(`uuid_generate_v4() as ${nodeDefName}_uuid`)
+
+          if (NodeDef.isCode(nodeDef)) {
+            await this._fetchCategoryItemsByNodeDef(nodeDef)
+          }
+        }
+
+        const selectData = dbGetQuery(schema, fromTables.join(' '), fields.join(', '), [
+          `${DataTable.colNameRecordCycle} = '${cycle}'`,
+        ])
         const setEntityData = setVar(NodeDef.getName(entityDef), selectData)
         await this.appendContent(setEntityData)
       }
     }
 
     return this
   }
+
+  /**
+   * Fetch category items and store them in a data frame called 'categoryName'
+   */
+  async _fetchCategoryItemsByNodeDef(nodeDef) {
+    const survey = this.rChain.survey
+    const defaultLang = R.pipe(Survey.getSurveyInfo, Survey.getDefaultLanguage)(survey)
+    const surveyId = Survey.getId(survey)
+
+    const nodeDefName = NodeDef.getName(nodeDef)
+    const category = Survey.getCategoryByUuid(NodeDef.getCategoryUuid(nodeDef))(survey)
+    const level = Category.getLevelByIndex(0)(category)
```

Comment:
don't you need to fetch the latest level?

Reply:
calculation attributes don't have a parent code attribute, they always use the first level of the category... isn't it?

Reply:
a category can have multiple levels. can't it? and the code attribute is linked to the last level of that category. or am i missing something?

Reply:
a code attribute is linked to the first level of the category, unless it has a parent code attribute specified and the analysis attributes cannot have a parent associated

Reply:
steeeee sorry, you are right !

Reply:
...you owe me a coffee ;)

Reply:
sure!!! :p